torch/serialization.py

```diff
@@ -1268,15 +1272,16 @@ def persistent_load(saved_id):
 
     deserialized_storage_keys = pickle_module.load(f, **pickle_load_args)
 
-    offset = f.tell() if f_should_read_directly else None
-    for key in deserialized_storage_keys:
-        assert key in deserialized_objects
-        typed_storage = deserialized_objects[key]
-        typed_storage._untyped_storage._set_from_file(
-            f, offset, f_should_read_directly,
-            torch._utils._element_size(typed_storage.dtype))
-        if offset is not None:
-            offset = f.tell()
+    if torch._guards.detect_fake_mode(None) is None:
```

Comment:
It's quite un-satisfying to have to do this kind of special casing.
Is the rationale behind this something like:
- We cannot override Storage-level constructor/functions via modes.
- Storage is a niche concept that we don't want to generalize too much, so we do not want to add such override.
- We are happy with the few systems using Storage to be special cased for any override that needs to happen.
- We are happy with this change not affecting Tensor serialized through numpy (anything without a storage + mtia, xla, ort as of today)

Reply:
Yes, exactly, to not special case, we would have to have some sort of Storage handling for fake mode.

The other strategy is to force the logic here to indirectly generate storages via tensors. Which may be feasible.

Reply:
Ok, That sounds fine if this is the only place. If we have more modes that need adding here or we need this special casing in other places, I think it would be worth working on the Tensor-based refactor.